ci-operator/step-registry/gather/proxy/gather-proxy-commands.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+set -o nounset
+set -o errexit
+set -o pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
+
+if test ! -f "${SHARED_DIR}/proxyregion"
+then
+	echo "No proxyregion, so unknown AWS region, so unable to tear down ."
+	exit 0
+fi
+
+REGION="$(cat "${SHARED_DIR}/proxyregion")"
+PROXY_NAME="${NAMESPACE}-${JOB_NAME_HASH}"
+
+# cleaning up after ourselves
+aws --region "${REGION}" s3 rb "s3://${PROXY_NAME}" --force
```

Comment:
Do we want a no-such-bucket guard?  Testing locally:

```console
$ aws s3 rb s3://mybucket-which-does-not-exist --force
fatal error: An error occurred (NoSuchBucket) when calling the ListObjectsV2 operation: The specified bucket does not exist

remove_bucket failed: Unable to delete all objects in the bucket, bucket will not be deleted.
$ echo $?
255
```

If the bucket doesn't exist, we probably don't have the stacks either.  But it would be slightly distracting to see this step fail (by exiting non-zero) when there were no proxy resources to clean up.  I dunno, I guess I'm fine dying with `NoSuchBucket`, unless folks can think of a convenient, race-free way to implement a guard.